src/providers/wms/qgswmsprovider.cpp

```diff
@@ -380,6 +381,34 @@ QString QgsWmsProvider::getLegendGraphicUrl() const
         break;
     }
   }
+  if ( mSettings.mIgnoreGetMapUrl )
+  {
+    if ( url.isEmpty() )
+    {
+      QgsDebugMsgLevel( QStringLiteral( "converting getLegendGraphicUrl is %1" ).arg( url ), 9 );
+      // rewrite the URL if the one in the capabilities document is incorrect
+      // strip every thing after the ? from the base url
+      const QStringList parts = mSettings.mBaseUrl.split( QRegularExpression( "\\?" ) );
+      const QString base = parts.isEmpty() ? mSettings.mBaseUrl : parts.first();
+      // and strip everything before the `rest` element (at least for GeoServer)
+      const int index = url.length() - url.lastIndexOf( QStringLiteral( "rest" ) ) + 1; // +1 for the /
```

Comment:
What exactly is the "rest" element? Do we have a problem if we don't have it int he url?

Reply:
Actually, this is in a condition `url.isEmpty()`, won't that always be empty?